src/box/sql/func.c

```diff
@@ -306,6 +306,11 @@ func_lower_upper(struct sql_context *ctx, int argc, const struct Mem *argv)
 		return mem_set_str0_static(ctx->pOut, "");
 	const char *str = arg->z;
 	int32_t len = arg->n;
+	if (len > SQL_MAX_LENGTH) {
+		diag_set(ClientError, ER_SQL_TOO_BIG_MEM_VALUE);
+		ctx->is_aborted = true;
+		return;
```

Comment:
This hunk and many others failing on `SQL_MAX_LENGTH` aren't covered by any tests. This is bad because we can't guarantee nothing bad happens if we hit any of those code paths.

Also, I don't like that we check the length in so many places. We can easily forget to add the check to a new places. Actually, I'm not sure you haven't missed a place where the check is required.

I see two ways to improve on this:
1. Check the length only when the mem is set (`mem_copy_bin` and similar functions).
2. Use `size_t` for string length instead of `int` and drop the checks completely or leave them only for user input.

I'd prefer the second way.

Still, we must fix the security issue ASAP so in the scope of this PR I'd fix only those places where we may leak data on invalid input or crash to close the referenced ticket. Refactoring required to make the code neat and clean can be done later.

Reply:
Understood. Reworked.